### DISCUSSION TITLE
Change mop mode to Deep on Roborock s7

### DISCUSSION THREAD

**Discussion by Brouwertje** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#581] [Q&A] [answered]
Hi!

Can't find anything on this subject.
How can i change the mop mode using HA. 

I can change it in de app. Fan speed is no problem, but can't change the MOP mode.

**Comment by PiotrMachowski** [comment]
You can try to figure it out by using `vacuum.send_command` service and [these commands](https://github.com/marcelrv/XiaomiRobotVacuumProtocol#vacuum-commands)

**Reply by Brouwertje** [reply]
Thanks! I can't figure it out :( The Roborock integration exposes mop_mode. 
Is it possible that your integration will be able to read the mop_mode and change it in the furture?

![afbeelding](https://user-images.githubusercontent.com/89314233/236305637-0d41b0cf-8dda-4b0f-8570-c8dbf4f8470c.png)

**Reply by PiotrMachowski** [reply]
My card can use almost any attribute/service. I just need a clear specification how to use them

**Reply by Brouwertje** [reply]
I'm not sure if this helps, but the roborock integration also shows mop_mode_list and mop_intensity_list

![afbeelding](https://user-images.githubusercontent.com/89314233/236318898-2939879f-8ffd-4aa2-bdf3-2a142aab92e6.png)

There are also two additional service calls available it seems. I'm trying to figure out how to use those in HA by using your integration (I use the Xiamio app and a roborock s7. Not using the Roborock app)

![afbeelding](https://user-images.githubusercontent.com/89314233/236319106-2389b29e-f576-428d-8c5b-0b766612c20e.png)

**Reply by Brouwertje** [reply]
I just realised that this is maybe a ''problem'' in the app i'm using, and not something you can add. Maybe the Xiaomi Miio app just doesn't expose the mop modes and the Roborock app does.

**Reply by PiotrMachowski** [reply, marked as answer]
Yeah, Xiaomi Miio integration doesn't seem to expose them (I'm not 100% sure though - my model doesn't have such features). I can add them to the Roborock platform though

**Reply by Brouwertje** [reply]
Cool, thanks a lot!
